### PR TITLE
added service principal check

### PR DESCRIPTION
**Pull request by cpareek**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPO-12956

### Change description ###

Added check for the Service Principal, it will run AZ cli command in shell script and then if there are any secrets expiring, it will notify plat ops team.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: scripts/service-principals.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+
+AZ_APP_RESULT=$( az ad app list --all --query "[?passwordCredentials[?endDateTime > '${TODAY_DATE}' && endDateTime < '${CHECK_DATE}']].{displayName:displayName, appId:appId, createdDateTime:createdDateTime, passwordCredentials:passwordCredentials[*].{displayName:displayName,endDateTime:endDateTime}}" --output json )
+
+#AZ_APP_RESULT=$( az ad app list --all --query "[?passwordCredentials[?endDateTime > '2023-03-28' && endDateTime < '2023-10-14']].{displayName:displayName, appId:appId, createdDateTime:createdDateTime, passwordCredentials:passwordCredentials[*].{displayName:displayName,endDateTime:endDateTime}}" --output json )
+
+AZ_APP_COUNT=$(jq -r '. | length' <<< "${AZ_APP_RESULT}")
+
+if [[ $AZ_APP_COUNT == 0 ]]; then
+    echo "None Service Principals Secrets are expiring in $CHECK_DAYS days"
```
Comment (adusumillipraveen) [inline scripts/service-principals.sh:17]: we should still append slack message in this case with green light.

File: scripts/service-principals.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+fi
+
+printf "\n:azure-826: <https://portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/RegisteredApps|_*Service Principal Secrets Status*_> \n\n" >> slack-message.txt
+
+echo "$AZ_APP_RESULT" | jq -c -r '.[]'  | while read i; do
+    displayName=$(jq -r '.displayName' <<< "$i")
+    appId=$(jq -r '.appId' <<< "$i")
+    APP_URL="https://portal.azure.com/#view/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/~/Credentials/appId/$appId"
+    printf "\n:yellow_circle: <$APP_URL|_* $displayName *_> Secret Expiring within next $CHECK_DAYS days " >> slack-message.txt
+done
```
Comment (adusumillipraveen) [inline scripts/service-principals.sh:28]: Looks great, probably make it 🔴 when it is expiring in less than 7 days?

File: scripts/service-principals.sh (hunk 1 above)
Comment (cpareek) [reply scripts/service-principals.sh]: done

File: scripts/service-principals.sh (hunk 2 above)
Comment (cpareek) [reply scripts/service-principals.sh:39]: sure-  make sense, done

File: scripts/service-principals.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+#AZ_APP_RESULT=$( az ad app list --all --query "[?passwordCredentials[?endDateTime > '2023-03-28' && endDateTime < '2023-04-29']].{displayName:displayName, appId:appId, createdDateTime:createdDateTime, passwordCredentials:passwordCredentials[?endDateTime > '2023-03-28' && endDateTime < '2023-04-29'].{displayName:displayName,endDateTime:endDateTime}}" --output json )
+
+AZ_APP_COUNT=$(jq -r '. | length' <<< "${AZ_APP_RESULT}")
+
+if [[ $AZ_APP_COUNT == 0 ]]; then
+    printf "\n:green_circle: None Service Principals Secrets are expiring in $CHECK_DAYS days \n\n" >> slack-message.txt
+    exit 0
+fi
+
+printf "\n:azure-826: <https://portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/RegisteredApps|_*Service Principal Secrets Status*_> \n" >> slack-message.txt
```
Comment (adusumillipraveen) [inline scripts/service-principals.sh:21]: this should probably move above line 16 , with current code  heading will come after green message .

**Pull request merged**